src/ray/raylet/node_manager.cc

```diff
@@ -3306,6 +3306,9 @@ void NodeManager::HandleGetNodeStats(const rpc::GetNodeStatsRequest &node_stats_
     return;
   }
   for (const auto &worker : all_workers) {
+    if (worker->IsDead()) {
```

Comment:
Does it cause line3332(send_reply_callback(Status::OK(), nullptr, nullptr);) not to be called?

Reply:
I prefer to add a default parameter to `GetAllRegisteredWorkers` and `GetAllRegisteredDrivers` to filter the worker and driver of alive (not filtered by default).

Reply:
Sounds good. Let me fix this.

Reply:
Shouldn't dead workers be filtered from those calls anyways? When do they actually get removed?

Reply:
It doesn't look like it. It only returns "Registered" workers, and there seem to be cases where there are dead registered workers. Maybe when the actor is marked as dead and deregistered are different. After I introduced this line, I saw some core worker stats failures logs from raylet have disappeared.